x-pack/plugins/observability_solution/observability_shared/common/entity/entity_types.ts

```diff
@@ -5,7 +5,25 @@
  * 2.0.
  */
 
-export enum EntityType {
-  HOST = 'host',
-  CONTAINER = 'container',
-}
+const createKubernetesEntity = <T extends string>(base: T) => ({
+  ecs: `kubernetes_${base}_ecs` as const,
+  semconv: `kubernetes_${base}_semconv` as const,
+});
+
+export const ENTITY_TYPES = {
+  HOST: 'host',
+  CONTAINER: 'container',
+  SERVICE: 'service',
+  KUBERNETES: {
+    CLUSTER: createKubernetesEntity('cluster'),
+    CONTAINER: createKubernetesEntity('container'),
+    CRONJOB: createKubernetesEntity('cron_job'),
+    DAEMONSET: createKubernetesEntity('daemon_set'),
+    DEPLOYMENT: createKubernetesEntity('deployment'),
+    JOB: createKubernetesEntity('job'),
+    NAMESPACE: createKubernetesEntity('namespace'),
+    NODE: createKubernetesEntity('node'),
+    POD: createKubernetesEntity('pod'),
+    STATEFULSET: createKubernetesEntity('stateful_set'),
+  },
```

Comment:
IMHO we should not type these.

Reply:
This is mostly needed to properly map an entity type with its detail view.